nucypher/policy/conditions/time.py

```diff
@@ -27,30 +25,32 @@ def make(self, data, **kwargs):
             return TimeCondition(**data)
 
     def __repr__(self) -> str:
-        r = f'{self.__class__.__name__}(timestamp={self.return_value_test.value})'
+        r = f"{self.__class__.__name__}(timestamp={self.return_value_test.value}, chain={self.chain})"
         return r
 
     def __init__(
         self,
         return_value_test: ReturnValueTest,
+        chain: int,
         method: str = METHOD,
         name: Optional[str] = None,
     ):
         if method != self.METHOD:
             raise InvalidCondition(
                 f"{self.__class__.__name__} must be instantiated with the {self.METHOD} method."
             )
-        self.return_value_test = return_value_test
-        self.name = name
+        super().__init__(
+            chain=chain, method=method, return_value_test=return_value_test, name=name
+        )
 
-    @property
-    def method(self):
-        return self.METHOD
+    def validate_method(self, method):
+        return method
 
     @property
     def timestamp(self):
         return self.return_value_test.value
 
-    def verify(self, *args, **kwargs) -> Tuple[bool, float]:
-        eval_time = time.time()   # system  clock
-        return self.return_value_test.eval(data=eval_time), eval_time
+    def _execute_call(self, parameters: List[Any]) -> Any:
+        """Execute onchain read and return result."""
+        latest_block = self.w3.eth.get_block("latest")
```

Comment:
In order to implement multicall, this should be replaced by a call to getCurrentBlockTimestamp() in the mds1/multicall contract. But this can be addressed in #3051.

Reply:
👍 . Indeed, let's do it as part of the `multicall` work.

I'm interested to look more deeply at the use of `multicall`  in `nucypher/nucypher`, I guess my uninformed expectation was that a condition object perhaps can returns a `Call` object for its intended condition action, that are collected and then executed as a single multi-call call (pun intended 😄 ).